nix/haskell-extra.nix

```diff
@@ -4,55 +4,76 @@
-  # https://github.com/input-output-hk/haskell.nix/issues/422 is fixed
-  # Also need to pick a version that builds properly
-  cabal-install = pkgs.haskell-nix.hackage-package {
-    name = "cabal-install";
-    version = "3.0.0.0";
+{ pkgs, index-state, checkMaterialization, sources }:
+pkgs.haskell-nix.tools {
+  cabal = {
+    version = "3.2.0.0";
     inherit index-state checkMaterialization;
     # Invalidate and update if you change the version or index-state
-    plan-sha256 = "08zkccwygm4g83chyiwbskkjfclm22vmhbx2s2rh0lvjkclqy6qc";
+    plan-sha256 = "034rggaixxhji42sb473w4kzwxr856kw393q3sjc4nsz76xpkkfl";
   };
-  stylish-haskell = pkgs.haskell-nix.hackage-package {
-    name = "stylish-haskell";
+  stylish-haskell = {
     version = "0.10.0.0";
     inherit index-state checkMaterialization;
     # Invalidate and update if you change the version or index-state
-    plan-sha256 = "0vdv4jjabblfn41nxpjgsyvy3yrgd4k2p0s39hhbifllga2ngzds";
+    plan-sha256 = "114npk6hjgcfa95fz8r28w6lxak4rslfvh9caiwmwrkgd8v3nmaz";
   };
-  hlint = pkgs.haskell-nix.hackage-package {
-    name = "hlint";
+  hlint = {
     version = "2.2.11";
     inherit index-state checkMaterialization;
     # Invalidate and update if you change the version or index-state
-    plan-sha256 = "0vmf2wzc2b9h4cxxj0mwpza9dy23n7dxadj6x7xaf8p9pmcmmmd5";
+    plan-sha256 = "1mppmhhfqsnwigg3apj43ylc6zc7zqyvnsimwbnxwicvir2xzdqm";
   };
-  haskell-language-server =
-    let hspkgs = pkgs.haskell-nix.cabalProject {
-        src = pkgs.fetchFromGitHub {
-          owner = "haskell";
-          repo = "haskell-language-server";
-          rev = "2186df00a9414c640fba1ae2acc3d9aa21ab6e4c";
-          sha256 = "0qh41jbf1a697l8wf48zmfs6vf08gijb0w42h26nvimcgc5dkh9a";
-          fetchSubmodules = true;
-        };
-        lookupSha256 = { location, tag, ... } : {
-          "https://github.com/wz1000/shake"."fb3859dca2e54d1bbb2c873e68ed225fa179fbef" = "0sa0jiwgyvjsmjwpfcpvzg2p7277aa0dgra1mm6afh2rfnjphz8z";
-          "https://github.com/peti/cabal-plan"."894b76c0b6bf8f7d2f881431df1f13959a8fce87" = "06iklj51d9kh9bhc42lrayypcpgkjrjvna59w920ln41rskhjr4y";
-          }."${location}"."${tag}";
-        inherit index-state checkMaterialization;
-        # Invalidate and update if you change the version
-        plan-sha256 = "0v9r3d11y595w7zrn7zrcxbcvjph5q8qj4fnig4sff99wcm038ab";
-        compiler-nix-name = "ghc883";
-        modules = [{
-          # Tests don't pass for some reason, but this is a somewhat random revision.
-          packages.haskell-language-server.doCheck = false;
-        }];
-      };
-    in hspkgs.haskell-language-server;
+  # This fails for packages that use plutus due to missing unfoldings
+  ghcide = {
+    version = "0.2.0";
+    compiler-nix-name = "ghc883";
+    index-state = "2020-06-02T00:00:00Z";
+    plan-sha256 = "04l2ihni8ccbnb99apsainmfww6swavxg8vw4h4cg966lcwayndh";
+  };
+} // {
+  haskell-language-server = (pkgs.haskell-nix.cabalProject {
```

Comment:
Maybe we should just push the ghcide fork into this so we just have the one ide tool and the same one will work for us when working on plutus?

Reply:
I think the object-code branch of ghcide has poor performance and I am not sure how to fix it.  It is also a bit out of date and I am not sure what other features it lacks (it may be more recent than the version used by hls though).

Reply:
Performance issues seem mostly be related to having multiple packages in the `cabal.project` file.

Reply:
Maybe we should just wait until Matthew finishes it then :laughing: